### PR TITLE
Eslint and lint fixes

### PR DESCRIPTION
**Pull request by raihanul-2k15**
### What's done

- Turned on most rules from airbnb config
- Fixed errors generated by those lint rules

**Note**: Couldn't fix two errors generated by (possibly buggy) ESLint rule
File: `./src/extensions/ConflictFixer/EventBody2EventBodyFloater.ts`
Rule: `consistent-return`

Some googling tells me that `consistent-return` isn't well suited for typescript. Checked a GitHub issue and a SO thread having similar issues. If you're OK disabling this rule, let me know and I'll push new code to this PR.

### Merge suggestion

Consider merging this PR with a squash. So all commits are compressed into one big merge commit.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,97 @@
+{
+    "extends": ["airbnb-base", "airbnb-typescript/base"],
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "project": "./tsconfig.json"
+    },
+    "plugins": [
+        "@typescript-eslint"
+    ],
+    "rules": {
```
Comment (pea3nut) [inline .eslintrc.json:10]: Why do we need so many rules?

Comment (raihanul-2k15) [reply .eslintrc.json:10]: The rules that say "error" can be deleted later, as they are enforced by airbnb already. 
The rules that are off is overriding the airbnb rules and turning them off.

Comment (pea3nut) [reply .eslintrc.json:10]: Why do we need to turn off some rules?

Comment (raihanul-2k15) [reply .eslintrc.json:10]: These rules feels like stupid rules to me

- @typescript-eslint/return-await
- func-names
- function-paren-newline
- class-methods-use-this
- arrow-parens
- object-curly-newline
- no-plusplus
- import/order
- no-console
- no-bitwise
- no-mixed-operators
- no-await-in-loop
- no-continue
- no-labels
- no-restricted-globals
- no-restricted-syntax

These ones are controversial. Need some discussion:
- import/no-cycle
- no-param-reassign
- no-lonely-if
- new-cap

These ones I feel like should be turned into warnings:
- @typescript-eslint/no-use-before-define
- @typescript-eslint/no-unused-expressions
- @typescript-eslint/no-unused-vars

This one I forgot to fix:
- @typescript-eslint/lines-between-class-members

This one needs discussion:
- @typescript-eslint/type-annotation-spacing

**Pull request merged**